src/core/filesystem.cc

```diff
@@ -895,7 +899,9 @@ Status
 ASFileSystem::CheckClient()
 {
   if (client_ == nullptr) {
-    return Status(Status::Code::INTERNAL, "blob client initialize failed.");
+    return Status(
+        Status::Code::INTERNAL,
+        "Unable to create Azure filesystem client. Check account credentials.");
```

Comment:
Just curious, if the client_ is nullptr then does that definitely means problem with account credentials? If not, can we get finer details of the error of why the client creation failed?

Reply:
The client is null if not initialized. This could be for many reasons. It is not possible to get the exact reason for the failure. Meanwhile even when credentials are valid (need not be correct for that specific blob storage) the client will be initialized correctly.